pkg/client/unversioned/request.go

```diff
@@ -585,8 +594,17 @@ func (r *Request) Watch() (watch.Interface, error) {
 	if client == nil {
 		client = http.DefaultClient
 	}
+
+	time.Sleep(time.Duration(backoff.CalculateBackoff(url)) * time.Second)
 	resp, err := client.Do(req)
 	updateURLMetrics(r, resp, err)
+	if r.baseURL != nil {
+		if err != nil {
+			backoff.UpdateBackoff(r.baseURL.Path, err, 0)
```

Comment:
I think you meant to pass `r.baseURL.String()` here.

But you should just pass around `*url.URL` types instead. You just parse the strings back to them anyway.